bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java

```diff
@@ -518,8 +830,16 @@ public Type toType(Datapoint datapoint, byte[] data) {
      * @return the openHAB type (command or state) class or {@code null} if the datapoint type id is not supported.
      */
     static public Class<? extends Type> toTypeClass(String dptId) {
-        logger.trace("toTypeClass looking for dptId = " + dptId);
-        return dptTypeMap.get(dptId);
+        Class<? extends Type> ohClass = dptTypeMap.get(dptId);
+        if (ohClass == null) {
+            int mainNumber = getMainNumber(dptId);
+            if (mainNumber == -1) {
+                logger.debug("toType: couldn't convert dptID toTypeClass: {}.", dptId);
```

Comment:
This would be confusing to me if I saw it in output. I'd expect the value at the end to be a 'TypeClass' value, not a 'dptID'.

Reply:
Better now.